### PR TITLE
librdkafka: add 1.9.2 + modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
- restore `cmake_find_package` in generators, since find modules have precedence over config files (important for zlib, openssl and libcurl).
- drop `CONAN_PKG::` logic in old patches, to prepare for conan v2 transition
- robust link of zstd targets (zstd::zstd won't exist in `CMakeDeps` generator, it's either `zstd::libzstd_shared` or `zstd::libzstd_static`)

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]

**Comment by SpaceIm** [comment]
```
[HOOK - conan-center.py] post_package_info(): ERROR: [LIBRARY DOES NOT EXIST (KB-H054)] Component librdkafka::rdkafka library 'rdkafka' is listed in the recipe, but not found installed at self.cpp_info.libdirs. Make sure you compiled the library correctly. If so, then the library name should probably be fixed. Otherwise, then the component should be removed. (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H054) 
[HOOK - conan-center.py] post_package_info(): ERROR: [LIBRARY DOES NOT EXIST (KB-H054)] Component librdkafka::rdkafka++ library 'rdkafka++' is listed in the recipe, but not found installed at self.cpp_info.libdirs. Make sure you compiled the library correctly. If so, then the library name should probably be fixed. Otherwise, then the component should be removed. (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H054) 
ERROR: 
	ConanException: [HOOK - conan-center.py] post_package_info(): Some checks failed running the hook, check the output
```

what?

**Comment by SpaceIm** [comment]
Can't reproduce locally with latest hooks. I'm running conan 1.51.0, it's conan 1.49.0 in CI?

File: recipes/librdkafka/all/conanfile.py [hunk 1]
```diff
@@ -75,51 +73,49 @@ def build_requirements(self):
...
 
     def build(self):
-        self._patch_sources()
-        with tools.run_environment(self):
-            cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        with tools_legacy.environment_append({"PKG_CONFIG_PATH": self.generators_folder}):
+            cmake.configure()
             cmake.build()
```
Comment (SpaceIm) [inline recipes/librdkafka/all/conanfile.py:109]: I've tested all shared (even pkgconf in build requirements) on macOS with `librdkafka:sasl=True` to test pkgconf, and it works like a charm thanks to `VirtualBuildEnv`, no SIP issue!

File: recipes/librdkafka/all/conanfile.py [hunk 2]
```diff
@@ -75,51 +73,49 @@ def build_requirements(self):
...
 
     def package(self):
-        self.copy(pattern="LICENSES.txt", src=self._source_subfolder, dst="licenses")
-        with tools.run_environment(self):
-            cmake = self._configure_cmake()
+        copy(self, "LICENSES.txt", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        with tools_legacy.environment_append({"PKG_CONFIG_PATH": self.generators_folder}):
```
Comment (SpaceIm) [inline recipes/librdkafka/all/conanfile.py:114]: @memsharded what is the canonical conan v2 way to inject an additional env var when we already have VirtualBuildEnv in generators?

File: recipes/librdkafka/all/conanfile.py [hunk 3]
```diff
@@ -66,60 +67,71 @@ def requirements(self):
...
+        if self._depends_on_cyrus_sasl:
+            self.tool_requires("pkgconf/1.7.4")
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+    def layout(self):
+        cmake_layout(self)
+        self.folders.source = "src"
+        self.folders.build = "build"
```
Comment (prince-chrismc) [inline recipes/librdkafka/all/conanfile.py:82]: These should be the defaults 🤔

**Review by prince-chrismc — COMMENTED** [review]
Looks super duper good 👍 

One tiny question

File: recipes/librdkafka/all/conanfile.py (hunk 3 above)
Comment (SpaceIm) [reply recipes/librdkafka/all/conanfile.py:82]: Good question

signature of cmake_layout is `cmake_layout(conanfile, generator=None, src_folder=".")`

So the default layout of `cmake_layout`:

```
build/
  <package_id>/
    source_code_files
    patches/
    build/<build_type>
source/
    source_code
    patches/
```

- source code and patches folder are mixed up, as well as build folder and source code (not a fully out of source build).
- and less important: `build/<build_type>` is unnecessarily long (might lead to path length issue on Windows)

So I propose instead this layout:

```
build/
  <package_id>/
    build/
    patches/
    src/source_code_files
source/
    patches/
    src/source_code_files
```

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**